Modules/SubModule_ASL/xASL_wrp_RealignASL.m

```diff
@@ -122,7 +122,11 @@ function xASL_wrp_RealignASL(x,bSubtraction)
 %         save('rp_ASL4D.txt','MotionAllTEs','-ascii')
 %     end  
     
-if nFrames>2 && bSubtraction && (numel(unique(x.Q.Initial_PLD))>1 || numel(unique(x.EchoTime))>1 || x.modules.asl.HadamardType~=0) %multiPLD or multiTE or Hadamard
+if nFrames>2 && bSubtraction && ...

```

Comment:
Not an entirely correct statement. it is multi-PLD OR multi-TE. So if echo-time is missing, you still want to check for multi-PLD...

We want this (but check if the brackets are ok;): 
nFrames>2 && bSubtraction && ( (isfield(x.Q,'Initial_PLD') && (numel(unique(x.Q.Initial_PLD))>1) || (isfield(x,'EchoTime') && numel(unique(x.EchoTime))>1) || (isfield(x.modules.asl,'TimeEncodedMatrixType') && x.modules.asl.TimeEncodedMatrixType~=0)) )

Reply:
Also - #859 seems to be solving a similar issue. Please apply the changes of #859 in this issue and mark #859 as duplicate.

Reply:
I thought Mathijs would add the multi-PLD check later on. This was just supposed to be a bugfix for the missing field checks, since the pipeline crashes without it.

Reply:
> 
> 
> Also - #859 seems to be solving a similar issue. Please apply the changes of #859 in this issue and mark #859 as duplicate.

@BeatrizPadrela already did this.